src/relay/charpane.ash

```diff
@@ -4412,26 +4410,234 @@ void bakeBricks() {
+	//counts the number of stars "*" in the current row of bricks
+	//used to decide what css width should be set to a brick or group of bricks.
+	float starCountHelper(string styleInfo) {
+		buffer bnum;
+		string[int] tokens = tokenize(styleInfo, $strings[*,%,px]);
+		float num = 1;
+		foreach i,s in tokens {
+			switch(s){
+			// a number before a "*" will return that number. Everything else will return 1.
+				case "%":
+				case "px":
+					num = 1;
+					break;
+				case "*":
+					return num;
 				default:
-					result.append(chitBricks[brick]);
+					num = to_float(s);
+			}
+		}
+		return 1;
+	}
+	
+	float total = 0;
+	foreach i,s in styleInfos {
+		total += starCountHelper(s);
+	}
+	return total;
+}
+
+string stylize(string styleInfo, int total_stars) {
+//returns an appropriate width style to a brick or group of bricks, based on user's chit layout preference
+	string[int] tokens = tokenize(styleInfo, $strings[*,%,px]);
+	string style = "";
+	float num = 0;
+	foreach i,s in tokens {
+		switch(s){
+			case "*":
+				style += "width:" + (100.0 * max(num,1)/total_stars) +"%;";
+				break;
+			case "px":
+				style += "min-width:" + num + "px;";
+				style += "max-width:" + num + "px;";
+				break;
+			case "%":
+				style += "width:" + num + "%;";
+				break;
+			default:
+				num = to_float(s);
+		}
+	}
+	if(style == "")
+		style = "width:" + (100.0 * max(num,1)/total_stars) +"%;";
+	
+	return 'style="' + style + '"';
+}
+
+buffer addGroup(string[int] rowHTMLs, string className) {
+//adds rows of bricks into a vertical group
+	buffer buff;
+	if(className != "")
+		append(buff, '<div class='+className+'>');
+		
+	foreach i,s in rowHTMLs {
+		append(buff, s);
+	}
+	
+	if(className != "")
+		append(buff, '</div>');
+
+	return buff;
+}
+
+buffer addRow(string[int] bricks, string[int] styleInfos) {
+//adds bricks or groups of bricks into horizontal rows
+	buffer buff;
+	string brick;
+	int totalStars = starCount(styleInfos) + count(bricks) - count(styleInfos);
+	//append(buff, '<div class="brick_row">');
+	foreach i,s in bricks {
+		brick = s;
+		if(chitBricks contains s) {
+			brick = chitBricks[s];
+		} else {
+			if(length(s) == 0 || substring(s,0,1) != "<")
+				brick = "";
+		}
+		if(length(brick) > 0) {
+			append(buff, '<div class="brick_holder" '+stylize(styleInfos[i],totalStars)+'>'+brick+'</div>');
+		}
+	}
+	//append(buff, '</div>');
+	return buff;
+}
+
+buffer addBricks (string[int] tokens) {
+//parses a chit.__.layout string and adding bricks in appropriate rows and columns,
+//with user-specified widths for the various bricks and groups of bricks
+//uses multi-dimensional arrays for tracking arrays at different parenthesis depth.
+	buffer result; 
+	string[int,int] bricks;
+	string[int,int] rowHTML;
+	string[int,int] styleInfo;
+	string brick;
+	boolean read_style = false;
+	int pLevel = 0;
+	int[int] i = {0:0};
+	int[int] j = {0:0};
+	foreach n,s in tokens {
+		if(read_style){
+			styleInfo[pLevel,i[pLevel]] = tokens[n];
```

Comment:
Wouldn't `s` work just fine instead of `tokens[n]`?

Reply:
Should be. This is a probably artefact of a previous version of the algorithm.